modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ClassPartialModifierAnalyzer.cs

```diff
@@ -0,0 +1,113 @@
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Document = Microsoft.CodeAnalysis.Document;
+
+namespace Godot.SourceGenerators
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class ClassPartialModifierAnalyzer : DiagnosticAnalyzer
+    {
+        private const string RuleId = "GD0001";
+        private const string Title = "Missing partial modifier on declaration of type";
+
+        private static readonly string MessageFormat =
+            "Missing partial modifier on declaration of type '{0}' that derives from '{1}'";
+
+        private static readonly string Description =
+            $"{Title}. Classes that derive from '{GodotClasses.GodotObject}' must be declared with the partial modifier.";
+
+        internal static readonly DiagnosticDescriptor Rule = new(
+            id: RuleId,
+            title: Title,
+            messageFormat: MessageFormat,
+            category: "Usage",
+            DiagnosticSeverity.Error,
+            isEnabledByDefault: true,
+            Description,
+            helpLinkUri: string.Format(Common.HelpLinkFormat, RuleId));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(
+                Rule);
```

Comment:
This line doesn't seem to be long enough to require a line-break here.

```suggestion
            => ImmutableArray.Create(Rule);
```